docs/examples/example_02_transport_svc.cpp

```diff
@@ -0,0 +1,305 @@
+
+        if (auto* const node_id_str = std::getenv("UAVCAN__NODE__ID"))
+        {
+            local_node_id_ = static_cast<NodeId>(std::stoul(node_id_str));
+        }
+        if (auto* const iface_address_str = std::getenv("UAVCAN__UDP__IFACE"))
+        {
+            iface_address_ = iface_address_str;
+        }
+
+        startup_time_ = executor_.now();
+    }
+
+    void TearDown() override
+    {
+        EXPECT_THAT(mr_.allocated_bytes, 0);
+        EXPECT_THAT(mr_.allocations, IsEmpty());
+        EXPECT_THAT(mr_.total_allocated_bytes, mr_.total_deallocated_bytes);
+    }
+
+    template <std::size_t Redundancy>
+    UdpTransportPtr makeUdpTransport(std::array<udp::IMedia*, Redundancy>& media_array, const NodeId local_node_id)
+    {
+        const std::size_t tx_capacity = 16;
+
+        auto maybe_transport = udp::makeTransport({mr_}, executor_, media_array, tx_capacity);
+        EXPECT_THAT(maybe_transport, VariantWith<UdpTransportPtr>(_)) << "Failed to create transport.";
+        auto udp_transport = cetl::get<UdpTransportPtr>(std::move(maybe_transport));
+
+        udp_transport->setLocalNodeId(local_node_id);
+
+        return udp_transport;
+    }
+
+    template <typename T, typename TxSession, typename TxMetadata>
+    static cetl::optional<AnyFailure> serializeAndSend(const T&          value,
+                                                       TxSession&        tx_session,
+                                                       const TxMetadata& metadata)
+    {
+        using traits = typename T::_traits_;
+        std::array<std::uint8_t, traits::SerializationBufferSizeBytes> buffer{};
+
+        const auto data_size = uavcan::node::serialize(value, buffer).value();
+
+        // NOLINTNEXTLINE
+        const cetl::span<const cetl::byte> fragment{reinterpret_cast<cetl::byte*>(buffer.data()), data_size};
+        const std::array<const cetl::span<const cetl::byte>, 1> payload{fragment};
+
+        return tx_session.send(metadata, payload);
+    }
+
+    void publishHeartbeat(const TimePoint now)
+    {
+        state_.tx_heartbeat_.transfer_id_ += 1;
+
+        const auto uptime         = now.time_since_epoch();
+        const auto uptime_in_secs = std::chrono::duration_cast<std::chrono::seconds>(uptime);
+
+        const uavcan::node::Heartbeat_1_0 heartbeat{static_cast<std::uint32_t>(uptime_in_secs.count()),
+                                                    {uavcan::node::Health_1_0::NOMINAL},
+                                                    {uavcan::node::Mode_1_0::OPERATIONAL}};
+
+        EXPECT_THAT(serializeAndSend(heartbeat,
+                                     *state_.tx_heartbeat_.msg_tx_session_,
+                                     TransferMetadata{state_.tx_heartbeat_.transfer_id_, now, Priority::Nominal}),
+                    Eq(cetl::nullopt))
+            << "Failed to publish heartbeat.";
+    }
+    void printHeartbeat(const MessageRxTransfer& rx_heartbeat)
```

Comment:
Subject id is fixed when you create a session

Reply:
Some more background on how we will primarily be using Libcyphal. We have a routing application whose only job is to send messages to their final destination. It does not care about the contents of the message and it does not deserialize messages. It will potentially have 100 subscriber objects. We will probably have a list of subscriber objects and it will need to loop over each one in the list and call its `receive` method. If there is a message available, the routing application will need to look up its subject ID and then route it to a higher-level app based on the subject ID. So, the subject ID is not a transport layer only thing for us. We will not have access to the DSDLs in our application and most of our messages do not use fixed subject IDs so they would not be filled in for us. We could maintain a map of Subscriber object to Subject ID, but I would prefer not have to use a map if we could add this to be part of the Metadata or provide a `getPortID` method in the Subscriber.

Reply:
I have 2 things to mention:
1. Along with `receive` method there is also `getParams()` - this is where you can get port id.
2. Now that we have executor and its callback mechanism, I'm thinking about adding to a RX session interface possibility to register "on receive" callback. I haven't discussed it with Pavel yet, but it seems that it will greatly simplify your use case (you don't need to loop over 100 sessions, instead your lambda will be called as soon as there is data; in such lambda you could also capture port id btw). Probably, presentation layer will also need such callback method.

Reply:
>I'm thinking about adding to a RX session interface possibility to register "on receive" callback.

I see no huge problem with this if it helps @lydia-at-amazon. I considered callbacks for the presentation layer but the plan was to postpone them until v1.1; I think they should be easy to add though:

![image](https://github.com/user-attachments/assets/1ed45361-0bf7-4cda-be7c-0379cafe711e)

Reply:
Thanks, Sergei and Pavel! @serges147 I saw the `getParams()` method but also saw that it was private and so didn't think we would have access to it at the Presentation Layer. The callback design sounds very helpful for our use case, but I am good with that coming in a separate PR from the initial Presentation Layer review.